### ISSUE TITLE
Analytics doesn't record the first sign-in when mandatorySignIn is true

### ISSUE THREAD

**Issue by phuctm97** [aws-amplify/amplify-js#8815]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication, Analytics

### Amplify Categories

auth, analytics

### Environment information

<details>

```
  System:
    OS: macOS 11.4
    CPU: (8) arm64 Apple M1
    Memory: 2.68 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.0 - ~/.nvm/versions/node/v14.17.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.17.0/bin/yarn
    npm: 7.15.0 - ~/.nvm/versions/node/v14.17.0/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Edge: 92.0.902.84
    Safari: 14.1.1
  npmGlobalPackages:
    npm: 7.15.0
    yarn: 1.22.10
```

</details>

### Describe the bug

When Amplify is configured with `mandatorySignIn: true`
Analytics doesn't record the first sign-in event.

```ts
Amplify.configure({
  Auth: {
    region: process.env.AWS_REGION,
    userPoolId: process.env.USER_POOL_ID,
    userPoolWebClientId: process.env.USER_POOL_CLIENT_ID,
    identityPoolId: process.env.IDENTITY_POOL_ID,
    authenticationFlowType: "CUSTOM_AUTH",
    process.env.AWS_REGION: true,
  },
  Analytics: {
    AWSPinpoint: {
      region: process.env.AWS_REGION,
      appId: process.env.PINPOINT_APP_ID,
    },
  },
});
```

On the second sign-in (when there is a user signed out, then someone sign in again during the same browser session), then the event is recorded correctly.

### Expected behavior

The first sign-in event should be recorded correctly.

### Reproduction steps

Start an Amplify app, configure the Identity Pool to not allow unauthenticated users, and configure the Amplify app like this:

```ts
Amplify.configure({
  Auth: {
    region: process.env.AWS_REGION,
    userPoolId: process.env.USER_POOL_ID,
    userPoolWebClientId: process.env.USER_POOL_CLIENT_ID,
    identityPoolId: process.env.IDENTITY_POOL_ID,
    authenticationFlowType: "CUSTOM_AUTH",
    process.env.AWS_REGION: true,
  },
  Analytics: {
    AWSPinpoint: {
      region: process.env.AWS_REGION,
      appId: process.env.PINPOINT_APP_ID,
    },
  },
});
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### Manual configuration

```ts
Amplify.configure({
  Auth: {
    region: process.env.AWS_REGION,
    userPoolId: process.env.USER_POOL_ID,
    userPoolWebClientId: process.env.USER_POOL_CLIENT_ID,
    identityPoolId: process.env.IDENTITY_POOL_ID,
    authenticationFlowType: "CUSTOM_AUTH",
    process.env.AWS_REGION: true,
  },
  Analytics: {
    AWSPinpoint: {
      region: process.env.AWS_REGION,
      appId: process.env.PINPOINT_APP_ID,
    },
  },
});
```

**Comment by cwomack** [comment]
Hello @phuctm97, and apologies for the delay in a response since you opened this issue.  Are you still experiencing this? 

If so, can you clarify what you mean by "doesn't record the first sign-in"?  Do you have a custom event being recorded upon hitting a sign in button of sorts?  Or are you referring to the `_session.start` event that Pinpoint captures out of the box? 

I only experienced missing events/errors when using the [scoped configuration](https://docs.amplify.aws/lib/client-configuration/configuring-amplify-categories/q/platform/js/#scoped-configuration-analytics) for Analytics specifically rather than importing/passing that into `Amplify.configure(awsExports)`.  Any chance you could try that locally to see if you get the same result? Either way I'm looking deeper into the scoped config and why it's creating an issue here. 

Related issue that was closed by bot and may need to be reopened: #3101

**Comment by cwomack** [comment]
Closing this as we have not heard back from you. If you are still experiencing this, please feel free to reply back and provide the information previously requested so we can re-open this issue and investigate further.  

Thank you!
